server/activate.go

```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/mattermost/mattermost-server/model"
+)
+
+// OnActivate is executed when the plugin is activated
+func (p *Plugin) OnActivate() error {
+	p.API.LogDebug("Activating plugin")
+
+	isCompatible, err := p.Helpers.CheckRequiredConfig(manifest.RequiredConfig, p.API.GetConfig())
```

Comment:
See comment in server PR, but the helpers have access to `p.API` directly so no need to pass the actual config. Seeing the example code here, I'm also wondering if we should have the `CheckRequiredConfig` return an error when the config is incompatible, allowing it to internalize the description of that error instead of forcing every plugin to do lines 19-23 for themselves.

Thoughts? It would return an error both when it failed to check, and when the config mismatched.